extensions/blocks/opentable/edit.js

```diff
@@ -0,0 +1,293 @@
+		if ( ! embedCode ) {
+			setErrorNotice();
+			return;
+		}
+
+		const scriptTagAttributes = embedCode.match( /< *script[^>]*src *= *["']?([^"']*)/i );
+		if ( ! scriptTagAttributes || ! scriptTagAttributes[ 1 ] ) {
+			setErrorNotice();
+			return;
+		}
+
+		let src = '';
+		if ( scriptTagAttributes[ 1 ].indexOf( 'http' ) === 0 ) {
+			src = new URL( scriptTagAttributes[ 1 ] );
+		} else {
+			src = new URL( 'http:' + scriptTagAttributes[ 1 ] );
+		}
+
+		if ( ! src.search ) {
+			setErrorNotice();
+			return;
+		}
+
+		const searchParams = new URLSearchParams( src.search );
+		let styleSetting = searchParams.get( 'theme' );
+		if ( searchParams.get( 'type' ) === 'button' ) {
+			styleSetting = searchParams.get( 'type' );
+		}
+
+		const newAttributes = {
+			rid: searchParams.getAll( 'rid' ),
+			iframe: Boolean( searchParams.get( 'iframe' ) ),
+			domain: searchParams.get( 'domain' ),
+			lang: searchParams.get( 'lang' ),
+			newtab: Boolean( searchParams.get( 'newtab' ) ),
+			style: styleSetting,
+		};
+
+		const validatedNewAttributes = getValidatedAttributes( defaultAttributes, newAttributes );
+		setAttributes( validatedNewAttributes );
+	};
+
+	const styleOptions = getStyleOptions( rid );
+	const styleValues = getStyleValues( rid );
+
+	const updateStyle = newStyle => {
+		setAttributes( { style: newStyle } );
+	};
+
+	const getTypeAndTheme = fromStyle =>
+		rid.length > 1
+			? [ 'multi', 'button' !== fromStyle ? fromStyle : 'standard' ]
+			: [
+					'button' === fromStyle ? 'button' : 'standard',
+					'button' === fromStyle ? 'standard' : fromStyle,
+			  ];
+
+	const blockPreview = styleOveride => {
+		const [ type, theme ] = getTypeAndTheme( styleOveride ? styleOveride : style );
+		return (
+			<>
+				<div className={ `${ className }-overlay` }></div>
+				<iframe
+					title={ `Open Table Preview ${ clientId }` }
+					src={ `https://www.opentable.com/widget/reservation/canvas?rid=${ join(
+						rid,
+						'%2C'
+					) }&type=${ type }&theme=${ theme }&overlay=false&domain=${ domain }&lang=${
+						lang && languageValues.includes( lang ) ? lang : 'en-US'
+					}&newtab=${ newtab }&disablega=true` }
+				/>
+			</>
+		);
+	};
+
+	const blockControls = (
+		<BlockControls>
+			{ ! isEmpty( rid ) && (
+				<Toolbar
+					isCollapsed={ true }
+					icon="edit"
+					label={ __( 'Type', 'jetpck' ) }
+					controls={ styleOptions.map( styleOption => ( {
+						title: styleOption.label,
+						isActive: styleOption.value === style,
+						onClick: () => updateStyle( styleOption.value ),
+					} ) ) }
+				/>
+			) }
+		</BlockControls>
+	);
+
+	const onPickerSubmit = input => {
+		if ( Array.isArray( input ) ) {
+			setAttributes( {
+				rid: input,
+				style: input.length > 1 && 'button' === style ? 'standard' : style,
+			} );
+		} else {
+			parseEmbedCode( input );
+		}
+	};
+
+	const restaurantPicker = ( pickerProps = {} ) => (
+		<RestaurantPicker rids={ rid } { ...pickerProps } />
+	);
```

Comment:
Is there a reason why we are not directly rendering this as:
```jsx
<RestaurantPicker rids={ rid } onChange={ onPickerSubmit } />

// and...

<RestaurantPicker
  rids={ rid }
  label={ __( 'Enter your restaurant name, OpenTable Restaurant ID or embed code', 'jetpack' ) }
  onSubmit={ onPickerSubmit }
/>
```
Looks simple enough to me! 🤔

Reply:
🚢